base/shared/src/main/scala/scalaz/data/identity.scala

```diff
@@ -19,11 +19,10 @@ trait IdentityInstances {
   })
 
   implicit final def identityEq[A](implicit A: Eq[A]): Eq[Identity[A]] =
-    instanceOf[EqClass[Identity[A]]](
-      (x, y) =>
-        (x, y) match {
-          case (Identity(a1), Identity(a2)) => A.equal(a1, a2)
-          case _                            => false
+    instanceOf(new EqClass[Identity[A]] {
```

Comment:
given removal of default implementation anonymous lambda syntax can be used there as well (also the superfluous `match` can be removed from original implementation).